Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/CqlCompiler.java

```diff
@@ -0,0 +1,255 @@
+package org.cqframework.cql.cql2elm;
+
+import org.antlr.v4.runtime.*;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.cqframework.cql.cql2elm.model.TranslatedLibrary;
+import org.cqframework.cql.cql2elm.preprocessor.CqlPreprocessorVisitor;
+import org.cqframework.cql.elm.tracking.TrackBack;
+import org.cqframework.cql.gen.cqlLexer;
+import org.cqframework.cql.gen.cqlParser;
+import org.fhir.ucum.UcumService;
+import org.hl7.elm.r1.Library;
+import org.hl7.elm.r1.Retrieve;
+import org.hl7.elm.r1.VersionedIdentifier;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CqlCompiler {
+    private Library library = null;
+    private TranslatedLibrary translatedLibrary = null;
+    private Object visitResult = null;
+    private List<Retrieve> retrieves = null;
+    private List<CqlTranslatorException> exceptions = null;
```

Comment:
With the separation of "Compiler" (which transforms CQL to ELM) and "Translator" (which converts ELM to a given serialization format) it seems most of these exceptions should be "CqlCompilerException". I'd only expect a "CqlTranslatorException" as part of serialization given the separation of the two concepts.

Reply:
That was my first impression as well. But then I realized that the class `CqlTranslatorException` is a catch-all term and parent class for many exceptions in this project. And since they are exposed to users, I believe the idea was to simply label exceptions from the CqlTranslator project.

Reply:
Coming back to this one. I converted CqlTranslatorException to CqlCompilerException again. I realized that the "Translated" exceptions were all on the code to compile dependencies of a given CQL, which before was still called "TranslatedLibraries".